src/AppRoutes.jsx

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { Routes, Route, Navigate, BrowserRouter } from "react-router-dom";
+import LandingPage from "./pages/LandingPage";
+import HomePage from "./pages/HomePage";
+import LoginPage from "./pages/LoginPage";
+import { useContext } from "react";
+import { UserContext } from "./contexts/UserContext";
+
+function AppRoutes() {
+  const { userInfo } = useContext(UserContext);
+  return (
+    <BrowserRouter>
+      <Routes>
+        <Route
+          path="/"
+          element={userInfo == null ? <LandingPage /> : <Navigate to="/homepage" />}
+        />
+        <Route
+          path="/login"
+          element={userInfo == null ? <LoginPage /> : <Navigate to="/homepage" />}
+        />
+        <Route path="/homepage" element={userInfo ? <HomePage /> : <Navigate to="/login" />} />
+      </Routes>
+    </BrowserRouter>
+  );
+}
+
+export default AppRoutes;
```

Comment:
It’s usually a good practice to wrap the entire app with BrowserRouter in a higher-level component, like index.js or App.js, to avoid re-rendering the router when switching routes. This also allows easier integration with other components like context providers.

Reply:
Thanks for the comment, I will move it to main.jsx